pkg/api/types.go

```diff
@@ -2088,3 +2088,83 @@ type RangeAllocation struct {
 	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
 	Data []byte `json:"data"`
 }
+
+// SecurityContextConstraints governs the ability to make requests that affect the SecurityContext
+// that will be applied to a container.
+type SecurityContextConstraints struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// AllowPrivilegedContainer determines if a container can request to be run as privileged.
+	AllowPrivilegedContainer bool `json:"allowPrivilegedContainer,omitempty" description:"allow containers to run as privileged"`
+	// AllowedCapabilities is a list of capabilities that can be requested to add to the container.
+	AllowedCapabilities []Capability `json:"allowedCapabilities,omitempty" description:"capabilities that are allowed to be added"`
+	// AllowHostDirVolumePlugin determines if the policy allow containers to use the HostDir volume plugin
+	AllowHostDirVolumePlugin bool `json:"allowHostDirVolumePlugin,omitempty" description:"allow the use of the host dir volume plugin"`
```

Comment:
I really think this is fine for now.  We don't have a great way to address
volume plugins in this way in the API and I fear it will be a debacle with
little pay-off if we try to invent one now.

On Tue, May 26, 2015 at 12:49 PM, Paul Weil <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/7893#discussion_r31054435>
> :
>
> > @@ -2088,3 +2088,83 @@ type RangeAllocation struct {
> >  	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
> >  	Data []byte `json:"data"`
> >  }
> > +
> > +// SecurityContextConstraints governs the ability to make requests that affect the SecurityContext
> > +// that will be applied to a container.
> > +type SecurityContextConstraints struct {
> > +	TypeMeta   `json:",inline"`
> > +	ObjectMeta `json:"metadata,omitempty"`
> > +
> > +	// AllowPrivilegedContainer determines if a container can request to be run as privileged.
> > +	AllowPrivilegedContainer bool `json:"allowPrivilegedContainer,omitempty" description:"allow containers to run as privileged"`
> > +	// AllowedCapabilities is a list of capabilities that can be requested to add to the container.
> > +	AllowedCapabilities []Capability `json:"allowedCapabilities,omitempty" description:"capabilities that are allowed to be added"`
> > +	// AllowHostDirVolumePlugin determines if the policy allow containers to use the HostDir volume plugin
> > +	AllowHostDirVolumePlugin bool `json:"allowHostDirVolumePlugin,omitempty" description:"allow the use of the host dir volume plugin"`
>
> Yes, Jordan suggested previously that we use an AllowedPlugins array.
> Let's discuss how to validate a request against various plugins and I can
> make that change.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/7893/files#r31054435>
> .
>